p2p/peer.go

```diff
@@ -296,28 +296,25 @@ const (
 // - Disconnected: peer disconnects, unmarking as connected and broadcasts a
 //   PeerStatusDown peer update.
 //
-// If we are connected to too many peers (more than MaxConnections), typically
-// because we have upgraded to higher-scored peers and need to shed lower-scored
-// ones, the flow is as follows:
-// - EvictNext: returns a peer ID to evict, marking peer as evicting.
-// - Disconnected: peer was disconnected, unmarking as connected and evicting,
-//   and broadcasts a PeerStatusDown peer update.
+// When evicting peers, either because peers are explicitly scheduled for
+// eviction or we are connected to too many peers, the flow is as follows:
+// - EvictNext: if marked evict and connected, unmark evict and mark evicting.
+//   If beyond MaxConnected, pick lowest-scored peer and mark evicting.
+// - Disconnected: unmark connected, evicting, evict, and broadcast a
+//   PeerStatusDown peer update.
 //
 // If all connection slots are full (at MaxConnections), we can use up to
 // MaxConnectionsUpgrade additional connections to probe any higher-scored
 // unconnected peers, and if we reach them (or they reach us) we allow the
-// connection and evict lower-scored peers. We mark the lower-scored peer as
+// connection and evict a lower-scored peer. We mark the lower-scored peer as
 // upgrading[from]=to to make sure no other higher-scored peers can claim the
 // same one for an upgrade. The flow is as follows:
-// - Accepted: if upgrade is possible, mark upgrading[from]=to and connected.
+// - Accepted: if upgrade is possible, mark connected and add to evict.
```

Comment:
what is added to evict? the lower-scored peer?

Reply:
Yes, will clarify.